packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmNotificationsTests.kt

```diff
@@ -211,4 +218,56 @@ class RealmNotificationsTests : NotificationTests {
             c.close()
         }
     }
+
+    // Will crash if we the collecting context doesn't follow up on processing notifications
+    // https://github.com/realm/realm-kotlin/issues/1147
+    @Ignore
```

Comment:
Could we enable this test and just capture the crash instead?

Reply:
No, because the `throw` isn't propagated to the recipient side so it is actually crashing the app. It should probably be propagated through `channel.close(exception)` instead to be signalled correctly, but need to experiment a bit with buffer control before I have a good feeling on the best way to handle backpressure without ruin our internal logic on events.

Reply:
I have tried to rework the way we report if we cannot deliver events on the notification flow and reworked the `ObjectBoundRealmResults` so that it doesn't fuse any external backpressure strategy and drops the internal events that we are heavily dependent on. Now failure to send events will close the flow with the exception issued in https://github.com/realm/realm-kotlin/pull/1132/files#diff-f05a939e8f402867f4467d8cabb245e48c8ebd28f3c5f92b752dab8122aa866dR142